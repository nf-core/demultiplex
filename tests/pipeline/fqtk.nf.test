nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "fqtk"
    tag "pipeline"

    test("FQTK") {

        when {
            params {
                input         = 'https://raw.githubusercontent.com/nf-core/test-datasets/demultiplex/samplesheet/1.3.0/fqtk-samplesheet.csv'
                demultiplexer = 'fqtk'
                outdir        = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions") },
                { assert workflow.trace.succeeded().size() == 105 },
                { assert snapshot(path("$outputDir/test/demux-metrics.txt")).match("fqtk") },
                { assert new File("$outputDir/test/unmatched_1.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/unmatched_2.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s10_1.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s10_2.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s11_1.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s11_2.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s12_1.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s12_2.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s13_1.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s13_2.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s14_1.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s14_2.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s15_1.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s15_2.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s16_1.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s16_2.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s17_1.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s17_2.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s18_1.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s18_2.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s19_1.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s19_2.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s1_1.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s1_2.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s20_1.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s20_2.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s21_1.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s21_2.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s22_1.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s22_2.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s23_1.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s23_2.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s24_1.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s24_2.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s2_1.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s2_2.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s3_1.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s3_2.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s4_1.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s4_2.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s5_1.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s5_2.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s6_1.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s6_2.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s7_1.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s7_2.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s8_1.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s8_2.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s9_1.fastp.fastq.gz").exists() },
                { assert new File("$outputDir/test/s9_2.fastp.fastq.gz").exists() }
            )
        }
    }
}
