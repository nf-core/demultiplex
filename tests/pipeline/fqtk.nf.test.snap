{
    "fqtk": {
        "content": [
            "demux-metrics.txt:md5,1d587fa959f9129155314cf531103347"
        ],
        "timestamp": "2023-10-17T08:06:53+0000"
    },
    "software_versions": {
        "content": [
            "{CUSTOM_DUMPSOFTWAREVERSIONS={python=3.11.4, yaml=6.0}, FALCO={falco=1.2.1}, FASTP={fastp=0.23.4}, FQTK={fqtk=0.2.1}, UNTAR={untar=1.30}, Workflow={nf-core/demultiplex=1.4.0dev}}"
        ],
        "timestamp": "2023-10-17T08:06:53+0000"
    }
}